sdk/core/src/main/java/com/klaviyo/coresdk/Klaviyo.kt

```diff
@@ -1,73 +1,146 @@
-    ) {
-        val request = TrackRequest(KlaviyoConfig.apiKey, event, customerProperties, properties)
-        processRequest(request)
-    }
+    fun setEmail(email: String): Klaviyo = this.setProfile(Profile().setEmail(email))
+
+    /**
+     * Assigns a phone number to the current internally tracked profile
+     *
+     * The SDK keeps track of current profile details to
+     * build analytics requests with profile identifiers
+     *
+     * This should be called whenever the active user in your app changes
+     * (e.g. after a fresh login)
+     *
+     * @param phoneNumber Phone number for active user
+     */
+    fun setPhoneNumber(phoneNumber: String): Klaviyo =
+        this.setProfile(Profile().setPhoneNumber(phoneNumber))
 
     /**
-     * Creates an identify request for the Klaviyo APIs and begins processing.
+     * Assigns an external ID to the current internally tracked profile
+     *
+     * The SDK keeps track of current profile details to
+     * build analytics requests with profile identifiers
+     *
+     * This should be called whenever the active user in your app changes
+     * (e.g. after a fresh login)
+     *
+     * @param id Phone number for active user
+     * @return
+     */
+    fun setExternalId(id: String): Klaviyo = this.setProfile(Profile().setExternalId(id))
+
+    /**
+     * Assign arbitrary attributes to the current profile by key
+     *
+     * This should be called when you collect additional data about your user
+     * (e.g. first and last name, or an address)
+     *
+     * @param propertyKey
+     * @param value
+     * @return
+     */
+    fun setProfileAttribute(propertyKey: ProfileKey, value: String): Klaviyo =
+        setProfile(Profile().also { it[propertyKey] = value })
+
+    /**
+     * Debounce timer for enqueuing profile API calls
+     */
+    private var timer: Clock.Cancellable? = null
+    private var pendingProfile: Profile? = null
+
+    /**
+     * Queues a request to identify profile properties to the Klaviyo API
+     *
+     * Any new identifying properties (externalId, email, phone) will be saved to the current
+     * internally tracked profile. Otherwise any identifiers missing from [profile] will be
+     * populated from the current internally tracked profile info.
+     *
      * Identify requests track specific properties about a user without triggering an event
      *
-     * @param properties A map of properties that define the user
+     * @param profile A map of properties that define the user
+     * @return
+     */
+    fun setProfile(profile: Profile): Klaviyo = apply {
+        // Update user identifiers in state
+        UserInfo.updateFromProfile(profile)
+
+        // Start or update a pending profile object for API call
+        pendingProfile = UserInfo.getAsProfile().merge(pendingProfile?.merge(profile) ?: profile)
+
+        timer?.cancel()
+        timer = Registry.clock.schedule(Registry.config.debounceInterval.toLong()) {
```

Comment:
Another nit: but thoughts on ripping the actual logic to enqueue the indentify request and process the debouncing into a private function? It just seems like it could be kept decoupled from the function of `setProfile`.

Otherwise I think this debounce logic on the requests looks great!

Reply:
Yeah I'm down for that.

Reply:
See https://github.com/klaviyo/klaviyo-android-sdk/pull/35 I extracted it which actually helped see how to make the other setters's code more readable